apps/nextjs-website/src/components/Header.tsx

```diff
@@ -106,21 +116,47 @@ const makeMegaHeaderProps = ({
     primary: link.label,
     secondary: link.sublinkGroups.map((sublinkGroup) => ({
       title: sublinkGroup.title,
-      items: sublinkGroup.sublinks.map(makeSublink),
+      items: sublinkGroup.sublinks.map((sublink) =>
+        makeSublink(sublink, locale),
+      ),
     })),
+    ...(link.ctaButton && {
+      ctaButton: {
+        ...link.ctaButton,
+        ...(link.ctaButton.icon && { startIcon: Icon(link.ctaButton.icon) }),
+      },
+    }),
   })),
 });
 
-const Header = (props: HeaderData['data']['attributes']['header'][0]) => {
-  const pathname = usePathname();
+const Header = (
+  props: HeaderData['data']['attributes']['header'][0] & {
+    locales: Array<Locale>;
+    defaultLocale: Locale;
+  },
+) => {
+  const pathname = usePathname() + '/';
   // eslint-disable-next-line
+  
+  const defaultLocale = props.defaultLocale;
+
+  const activeLocale = ['it/', 'en/', 'de/', 'fr/', 'sl/'].includes(
+    pathname.slice(1, 4),
+  )
+    ? (pathname.slice(1, 3) as Locale)
+    : defaultLocale;
+
   const menuType = props.__component;
```

Comment:
```suggestion
  const pathname = usePathname();
  // eslint-disable-next-line
  const menuType = props.__component;
```
No need to find the activeLocale, just use locale.